lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,881 @@
+    // This limitation is due to the fact that the GELF format does not specify the length of the
+    // message, so we have to read all the bytes from the message (datagram)
+    bytes_decoder: BytesDecoder,
+    state: Arc<Mutex<HashMap<u64, MessageState>>>,
+    timeout: Duration,
+    pending_messages_limit: Option<usize>,
+    max_chunk_length: Option<usize>,
+    max_message_length: Option<usize>,
+}
+
+impl ChunkedGelfDecoder {
+    /// Creates a new `ChunkedGelfDecoder`.
+    pub fn new(
+        timeout_secs: f64,
+        pending_messages_limit: Option<usize>,
+        max_chunk_length: Option<usize>,
+        max_message_length: Option<usize>,
+    ) -> Self {
+        Self {
+            bytes_decoder: BytesDecoder::new(),
+            state: Arc::new(Mutex::new(HashMap::new())),
+            timeout: Duration::from_secs_f64(timeout_secs),
+            pending_messages_limit,
+            max_chunk_length,
+            max_message_length,
+        }
+    }
+
+    /// Decode a GELF chunk
+    pub fn decode_chunk(
+        &mut self,
+        mut chunk: Bytes,
+    ) -> Result<Option<Bytes>, ChunkedGelfDecoderError> {
+        // Encoding scheme:
+        //
+        // +------------+-----------------+--------------+----------------------+
+        // | Message id | Sequence number | Total chunks |    Chunk payload     |
+        // +------------+-----------------+--------------+----------------------+
+        // | 64 bits    | 8 bits          | 8 bits       | remaining bits       |
+        // +------------+-----------------+--------------+----------------------+
+        //
+        // As this codec is oriented for UDP, the chunks (datagrams) are not guaranteed to be received in order,
+        // nor to be received at all. So, we have to store the chunks in a buffer (state field) until we receive
+        // all the chunks of a message. When we receive all the chunks of a message, we can concatenate them
+        // and return the complete payload.
+
+        // We need 10 bytes to read the message id, sequence number and total chunks
+        ensure!(
+            chunk.remaining() >= 10,
+            InvalidChunkHeaderSnafu { header: chunk }
+        );
+
+        let message_id = chunk.get_u64();
+        let sequence_number = chunk.get_u8();
+        let total_chunks = chunk.get_u8();
+
+        ensure!(
+            total_chunks > 0 && total_chunks <= GELF_MAX_TOTAL_CHUNKS,
+            InvalidTotalChunksSnafu {
+                message_id,
+                sequence_number,
+                total_chunks
+            }
+        );
+
+        ensure!(
+            sequence_number < total_chunks,
+            InvalidSequenceNumberSnafu {
+                message_id,
+                sequence_number,
+                total_chunks
+            }
+        );
+
+        if let Some(max_chunk_length) = self.max_chunk_length {
+            let chunk_length = chunk.remaining();
+            ensure!(
```

Comment:
Should we remove the state of the message id of this chunk? 

I don't know if it makes sense to keep the buffered chunks if we discars one, as the message would never complete

Reply:
I wonder if we should also do this for the `InvalidTotalChunks` error  for example, or in any other case..?

Reply:
Mmm, yeah, seemingly the message will never be complete in this scenario since one of the chunks is too big. I'd be ok with removing the message id. After thinking about it a bit more, I'm also not sure if users will need to limit the individual chunk size, rather than just the message size, so I'd also be ok if you wanted to drop the chunk max limit for now to simplify things until we need it and just leave max message length.